lib/src/widgets/adw/raw_target.dart

```diff
@@ -0,0 +1,156 @@
+import 'package:flutter/material.dart';
+
+/// Set of status that a [RawAdwTarget] widget can be at any given time.
+enum TargetStatus { enabled, active, hover, tapDown }
+
+typedef TargetColorBuilder = Color? Function(BuildContext, TargetStatus);
+
+typedef TargetWidgetBuilder = Widget Function(BuildContext, TargetStatus);
+
+/// Basic widget that can be used to create other more complex and customized
+/// buttons and other set of 'target' views.
+///
+/// It offers the ability to reconstruct sets of properties depending on the
+/// current state of the view.
+class RawAdwTarget extends StatefulWidget {
+  const RawAdwTarget({
+    Key? key,
+    this.padding = EdgeInsets.zero,
+    required this.builder,
+    this.onPressed,
+    required this.backgroundColorBuilder,
```

Comment:
Do we want this to be `@required`? What if the user simply wants the default background but still want to use the gesture mechanism given by `_status`? Does that seems like an unlikely use of the class?

Reply:
You're right here. It could be a use case where no color is needed, thus the `backgroundColorBuilder` parameter should be 'nullable'.

If so, no color will be painted in the background and it will be transparent.